eng/pipelines/pr.yml

```diff
@@ -0,0 +1,241 @@
+parameters:
+- name: RunUnitTestsOnWindows
+  displayName: Run unit tests on Windows
+  type: boolean
+  default: true
+- name: RunFunctionalTestsOnWindows
+  displayName: Run functional tests on Windows
+  type: boolean
+  default: true
+- name: RunUnitTestsOnLinux
+  displayName: Run unit tests on Linux
+  type: boolean
+  default: true
+- name: RunFunctionalTestsOnLinux
+  displayName: Run functional tests on Linux
+  type: boolean
+  default: true
+- name: RunUnitTestsOnMacOS
+  displayName: Run unit tests on MacOS
+  type: boolean
+  default: true
+- name: RunFunctionalTestsOnMacOS
+  displayName: Run functional tests on MacOS
+  type: boolean
+  default: true
+
+variables:
+  DOTNET_NOLOGO: 1
+  NUGET_EXPERIMENTAL_CHAIN_BUILD_RETRY_POLICY: 3,1000
+  SkipSigning: true
+  CI: true
+
+trigger:
+  branches:
+    include:
+    - dev
```

Comment:
Wouldn't we want to run CI on all PRs, not just the dev branch?  CI on release branches are useful, but moreso if we have some `dev-feature-branch`, we want to run CI so we can test PRs before the whole feature branch is ready to go into dev.

Reply:
I'm not sure, we currently have the PrivateDev branch set up to just build all branches that start with `dev-`.  These settings will only trigger on PRs to the `dev` branch as you say.  I'm fine with just triggering it for all branches or something like:
```yaml
pr:
  branches:
    include:
    - dev
    - release-*
    - feature-*
    - dev-feature-*
```

Definitely easy to change as we go as well.